packages/core/src/mesh/ModelMesh.ts

```diff
@@ -14,4 +66,800 @@ export class ModelMesh extends Mesh {
     super(engine);
     this.name = name;
   }
+
+  /**
+   * Upload Mesh Data to the graphics API.
+   * @param noLongerAccessible - Whether to access data later. If true, you'll never access data anymore (free memory cache)
+   */
+  uploadMeshData(noLongerAccessible: boolean): void {
+    const { _indices: _indicesArray, _vertexSlotChanged, _accessible } = this;
+    if (!_accessible) {
+      throw "Not allowed to access data while accessible is false.";
+    }
+
+    // Structure of vertex buffer has changed
+    if (_vertexSlotChanged || this._vertexCountChanged) {
+      const vertexElements = this._updateVertexElements();
+      const elementCount = this._elementCount;
+      const vertices = this._verticesFloat32;
+      if (!vertices || vertices.length !== elementCount * this._vertexCount) {
+        const vertices = new Float32Array(elementCount * this._vertexCount);
+        this._verticesUint8 = new Uint8Array(vertices.buffer);
+        this._verticesFloat32 = vertices;
+      }
+      this._vertexChangeFlag = ValueChanged.All;
+      this._updateVertices(vertices);
+      this._vertexBuffer = new Buffer(
```

Comment:
1: "this._vertexBuffer = new Buffer( " should include in "if (!vertices || vertices.length !== elementCount * this._vertexCount)" condiation. 
2: You should destroy verexbuffer if this._vertexBuffer is not null.

what do you think  write like this:
```typescript
    // Structure of vertex buffer has changed
    let vertices = this._verticesFloat32;
    if (_vertexSlotChanged || this._vertexCountChanged) {
      const vertexElements = this._updateVertexElements();
      const elementCount = this._elementCount;

      if (!vertices || vertices.length !== elementCount * this._vertexCount) {
        vertices && this._vertexBuffer.destroy();
        vertices = new Float32Array(elementCount * this._vertexCount);
        const vertexBuffer = new Buffer(
          this._engine,
          BufferBindFlag.VertexBuffer,
          vertices.byteLength,
          noLongerAccessible ? BufferUsage.Static : BufferUsage.Dynamic
        );

        this.setVertexBufferBinding(vertexBuffer, elementCount * 4);
        this._verticesUint8 = new Uint8Array(vertices.buffer);
        this._verticesFloat32 = vertices;
        this._vertexBuffer = vertexBuffer;
      }

      this.setVertexElements(vertexElements);
      this._vertexChangeFlag = ValueChanged.All;
      this._vertexSlotChanged = false;
      this._vertexCountChanged = false;
    }

    this._updateVertices(vertices);
    this._vertexBuffer.setData(vertices);
```

Reply:
+1